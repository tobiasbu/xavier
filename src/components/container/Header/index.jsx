import React from 'react';
import PropTypes from 'prop-types';

import Title from '../../basic/Title';

import useStyle from './style';

/**
 * Page header
 * @param {HeaderProps} props Header props.
 */
const Header = (props) => {
  const { children } = props;
  const classes = useStyle();

  return (
    <header className={classes.headerContainer}>
      <div className={classes.headerInner}>
        <Title size={3}>{children}</Title>
      </div>
    </header>
  );
};

/**
 * Header prop types
 * @type {HeaderProps}
 */
Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
};

export default Header;
