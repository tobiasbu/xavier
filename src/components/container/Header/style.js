import { createUseStyles } from 'react-jss';

const useStyle = createUseStyles({
  headerContainer: {
    position: 'relative',
    height: '100%',
    maxHeight: 161,
    '& h3': {
      margin: 0,
      lineHeight: 1,
    },
  },
  headerInner: {
    padding: '3.5rem 0 0',
    margin: 'auto',
  },
});

export default useStyle;
