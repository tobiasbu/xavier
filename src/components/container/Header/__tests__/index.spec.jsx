import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Header from '..';

describe('Header', () => {
  it('should render', () => {
    const component = shallow(
      <Header>
        xavier is cool and have super powers
      </Header>,
    );
    expect(toJSON(component)).toMatchSnapshot();
  });
});
