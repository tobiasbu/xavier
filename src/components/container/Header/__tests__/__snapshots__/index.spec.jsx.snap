// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<header
  className="headerContainer-0-2-1"
>
  <div
    className="headerInner-0-2-2"
  >
    <Title
      className=""
      size={3}
    >
      xavier is cool and have super powers
    </Title>
  </div>
</header>
`;
